test/extensions/filters/http/jwt_authn/group_verifier_test.cc

```diff
@@ -0,0 +1,537 @@
+#include "extensions/filters/http/jwt_authn/verifier.h"
+
+#include "test/extensions/filters/http/jwt_authn/mock.h"
+#include "test/extensions/filters/http/jwt_authn/test_common.h"
+#include "test/test_common/utility.h"
+
+#include "gmock/gmock.h"
+
+using ::envoy::config::filter::http::jwt_authn::v2alpha::JwtAuthentication;
+using ::google::jwt_verify::Status;
+using ::testing::NiceMock;
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace JwtAuthn {
+namespace {
+const char AllWithAny[] = R"(
+providers:
+  provider_1:
+  provider_2:
+  provider_3:
+rules:
+- match: { path: "/" }
+  requires:
+    requires_all:
+      requirements:
+      - requires_any:
+          requirements:
+            - provider_name: "provider_1"
+            - provider_name: "provider_2"
+      - provider_name: "provider_3"
+)";
+
+const char AnyWithAll[] = R"(
+providers:
+  provider_1:
+  provider_2:
+  provider_3:
+  provider_4:
+rules:
+- match: { path: "/" }
+  requires:
+    requires_any:
+      requirements:
+      - requires_all:
+          requirements:
+            - provider_name: "provider_1"
+            - provider_name: "provider_2"
+      - requires_all:
+          requirements:
+            - provider_name: "provider_3"
+            - provider_name: "provider_4"
+)";
+
+class GroupVerifierTest : public ::testing::Test {
+public:
+  void createVerifier() {
+    ON_CALL(mock_factory_, create(_, _, _))
+        .WillByDefault(
+            Invoke([&](const std::vector<std::string>&, const absl::optional<std::string>&, bool) {
+              return std::move(mock_auths_[idx++]);
```

Comment:
If we store the issuers for each involved auth.  the test code can assert on the used issuer too to make it more precious which providers are used in the test.